File: examples/python/sqs_delete_message.py
"""Elimina (confirma el procesamiento de) un mensaje de una cola SQS.

Uso: python3 sqs_delete_message.py <queue-url> <receipt-handle>
"""
import sys

import boto3

sqs = boto3.client(
    "sqs",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 3:
        raise SystemExit("Uso: python3 sqs_delete_message.py <queue-url> <receipt-handle>")
    queue_url, receipt_handle = sys.argv[1], sys.argv[2]

    sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
    print("Mensaje eliminado de la cola.")


if __name__ == "__main__":
    main()

File: examples/go/sqs_delete_message.go
// Elimina (confirma el procesamiento de) un mensaje de una cola SQS.
// Uso: go run sqs_delete_message.go <queue-url> <receipt-handle>
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/credentials"
	"github.com/aws/aws-sdk-go-v2/service/sqs"
)

func main() {
	if len(os.Args) < 3 {
		log.Fatal("Uso: go run sqs_delete_message.go <queue-url> <receipt-handle>")
	}
	queueURL := os.Args[1]
	receiptHandle := os.Args[2]

	ctx := context.Background()
	cfg, err := config.LoadDefaultConfig(ctx,
		config.WithRegion("us-east-1"),
		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "")),
	)
	if err != nil {
		log.Fatalf("Error cargando configuracion: %v", err)
	}
	client := sqs.NewFromConfig(cfg, func(o *sqs.Options) {
		o.BaseEndpoint = aws.String("http://localhost:4566")
	})

	_, err = client.DeleteMessage(ctx, &sqs.DeleteMessageInput{QueueUrl: aws.String(queueURL), ReceiptHandle: aws.String(receiptHandle)})
	if err != nil {
		log.Fatalf("Error eliminando el mensaje (el ReceiptHandle pudo haber expirado): %v", err)
	}
	fmt.Println("Mensaje eliminado de la cola.")
}
